hw/ip/otp_ctrl/data/BUILD

```diff
@@ -155,51 +172,66 @@ otp_json(
             items = {
                 # Enable bootstrap. See `hardened_bool_t` in
                 # sw/device/lib/base/hardened.h.
-                "OWNER_SW_CFG_ROM_BOOTSTRAP_DIS": "0x1d4",
+                "OWNER_SW_CFG_ROM_BOOTSTRAP_DIS": otp_hex(CONST.FALSE),
                 # Set to 0x739 to use the ROM_EXT hash measurement as the key
                 # manager attestation binding value.
-                "OWNER_SW_CFG_ROM_KEYMGR_ROM_EXT_MEAS_EN": "0x0",
+                "OWNER_SW_CFG_ROM_KEYMGR_ROM_EXT_MEAS_EN": otp_hex(0x0),
                 # Report errors without any redaction.
-                "OWNER_SW_CFG_ROM_ERROR_REPORTING": "0xe2290aa5",
+                "OWNER_SW_CFG_ROM_ERROR_REPORTING": otp_hex(CONST.SHUTDOWN.REDACT.NONE),
                 # Set the enables to kAlertEnableNone.
                 # See `alert_enable_t` in
                 # sw/device/silicon_creator/lib/drivers/alert.h
-                "OWNER_SW_CFG_ROM_ALERT_CLASS_EN": "0xa9a9a9a9",
-                # Set the esclation policies to kAlertEscalateNone.
+                "OWNER_SW_CFG_ROM_ALERT_CLASS_EN": otp_per_class_bytes(
+                    A = CONST.ALERT.NONE,
+                    B = CONST.ALERT.NONE,
+                    C = CONST.ALERT.NONE,
+                    D = CONST.ALERT.NONE,
+                ),
+                # Set the escalation policies to kAlertEscalateNone.
                 # See `alert_escalate_t` in
                 # sw/device/silicon_creator/lib/drivers/alert.h
-                "OWNER_SW_CFG_ROM_ALERT_ESCALATION": "0xd1d1d1d1",
-                # Set the classifiactions to kAlertClassX.
+                "OWNER_SW_CFG_ROM_ALERT_ESCALATION": otp_per_class_bytes(
+                    A = CONST.ALERT.ESC_NONE,
+                    B = CONST.ALERT.ESC_NONE,
+                    C = CONST.ALERT.ESC_NONE,
+                    D = CONST.ALERT.ESC_NONE,
+                ),
+                # Set the classifications to kAlertClassX.
                 # See `alert_class_t` in
                 # sw/device/silicon_creator/lib/drivers/alert.h
-                "OWNER_SW_CFG_ROM_ALERT_CLASSIFICATION": ["0x94949494"] * 80,
-                # Set the classifiactions to kAlertClassX. See `alert_class_t` in
+                "OWNER_SW_CFG_ROM_ALERT_CLASSIFICATION": otp_alert_classification(
+                    alert_list = EARLGREY_ALERTS,
+                    default = alert(
+                        dev = "X",
+                        prod = "X",
+                        prod_end = "X",
+                        rma = "X",
+                    ),
```

Comment:
This would be nice to read in tabular form, but according to https://github.com/bazelbuild/buildtools/issues/890, it seems like there is no good solution for getting buildifier to not format this.

If we remove the kwargs keywords, buildifier _will_ leave this on a single line, but the columns will be at the mercy of the length of the alert name:
```
"OWNER_SW_CFG_ROM_ALERT_CLASSIFICATION": otp_alert_classification(
    alert_list = EARLGREY_ALERTS,
    # The ordering is alert(prod, prod_end, dev, rma)
    default = alert("X", "X", "X", "X"),
    uart0_fatal_fault = alert("X", "X", "X", "X"),
    gpio_fatal_fault = alert("X", "X", "X", "X"),
    spi_device_fatal_fault = alert("X", "X", "X", "X"),
    i2c0_fatal_fault = alert("X", "X", "X", "X"),
    pattgen_fatal_fault = alert("X", "X", "X", "X"),
    rv_timer_fatal_fault = alert("X", "X", "X", "X"),
    otp_ctrl_fatal_macro_error = alert("X", "X", "X", "X"),
),
```

Have you got any ideas on making this look nice?

Reply:
One possibility is to splat a dictionary into the arguments of `otp_alert_classification` and express the alert names as strings with extra whitespace... and then to trim out the extra whitespace when processing:
```
"OWNER_SW_CFG_ROM_ALERT_CLASSIFICATION": otp_alert_classification(**{
    "alert_list": EARLGREY_ALERTS,
    # The ordering is alert(prod, prod_end, dev, rma)
    "default                   ": alert("X", "X", "X", "X"),
    "uart0_fatal_fault         ": alert("X", "X", "X", "X"),
    "gpio_fatal_fault          ": alert("X", "X", "X", "X"),
    "spi_device_fatal_fault    ": alert("X", "X", "X", "X"),
    "i2c0_fatal_fault          ": alert("X", "X", "X", "X"),
    "pattgen_fatal_fault       ": alert("X", "X", "X", "X"),
    "rv_timer_fatal_fault      ": alert("X", "X", "X", "X"),
    "otp_ctrl_fatal_macro_error": alert("X", "X", "X", "X"),
}), 
```

Reply:
Yeah I wrestled with this issue for a while today. Both the issue you linked above and bazelbuild/buildtools#806 seem to suggest that while the linting can be turned off, the auto-formatting is unavoidable. At the moment, I've actually just added an exception for this file for buildifier, but that's more of a workaround if we can't find anything.

I think the dictionary splat might be the way to go on this. If that seems okay, I can add the trimming and remove the buildifier exception.

Reply:
Unfortunately, splatting a dictionary with `**` appears to be [forbidden in BUILD files](https://bazel.build/concepts/build-files) with the following error: 
```
**kwargs arguments are not allowed in BUILD files. Pass the arguments in explicitly.
```

 If we really want to, we can add a dummy first argument to `alert` that's just there for spacing:
```
"OWNER_SW_CFG_ROM_ALERT_CLASSIFICATION": otp_alert_classification(
    alert_list = EARLGREY_ALERTS,
    # The ordering is alert("        ", prod, prod_end, dev, rma)
    default = alert("                   ", "X", "X", "X", "X"),
    gpio_fatal_fault = alert("          ", "X", "X", "X", "X"),
    i2c0_fatal_fault = alert("          ", "X", "X", "X", "X"),
    otp_ctrl_fatal_macro_error = alert("", "X", "X", "X", "X"),
    pattgen_fatal_fault = alert("       ", "X", "X", "X", "X"),
    rv_timer_fatal_fault = alert("      ", "X", "X", "X", "X"),
    spi_device_fatal_fault = alert("    ", "X", "X", "X", "X"),
    uart0_fatal_fault = alert("         ", "X", "X", "X", "X"),
),
```

or we would just disable buildifier formatting on this file. What do you think?

Reply:
Okay that sounds good